app/models/step_by_step_page.rb

```diff
@@ -5,6 +5,9 @@ class StepByStepPage < ApplicationRecord
   validates :slug, format: { with: /\A([a-z0-9]+-)*[a-z0-9]+\z/ }
   validates :slug, uniqueness: true
 
+  include ActiveModel::Validations
+  validates_with SlugValidator
```

Comment:
I think we can replace the separate slug validation lines with something like:
`validates :slug, format: { with: /\A([a-z0-9]+-)*[a-z0-9]+\z/ }, uniqueness: true, slug: true`

The `slug: true` bit works out that there's a validator called `SlugValidator`